components/cam/src/physics/cam/micro_p3.F90

```diff
@@ -896,222 +895,65 @@ SUBROUTINE p3_main(qc,nc,qr,nr,th_old,th,qv_old,qv,dt,qitot,qirim,nitot,birim,ss
-             nccol = rhofaci(i,k)*f1pr04*nc_incld(i,k)*eci*rho(i,k)*nitot_incld(i,k)
-          endif
-
-          ! for T > 273.15, assume cloud water is collected and shed as rain drops
-
-          if (qitot_incld(i,k).ge.qsmall .and. qc_incld(i,k).ge.qsmall .and. t(i,k).gt.zerodegc) then
-             ! sink for cloud water mass and number, note qcshed is source for rain mass
-             qcshd = rhofaci(i,k)*f1pr04*qc_incld(i,k)*eci*rho(i,k)*nitot_incld(i,k)
-             nccol = rhofaci(i,k)*f1pr04*nc_incld(i,k)*eci*rho(i,k)*nitot_incld(i,k)
-             ! source for rain number, assume 1 mm drops are shed
-             ncshdc = qcshd*1.923e+6_rtype
-          endif
+          call ice_droplet_collection(rho(i,k),t(i,k),rhofaci(i,k),&
+          f1pr04,qitot_incld(i,k),qc_incld(i,k),nitot_incld(i,k),nc_incld(i,k),&
+               qccol,nccol,qcshd,ncshdc)
 
           !....................
           ! collection of rain
-
-          ! here we multiply rates by air density, air density fallspeed correction
-          ! factor, collection efficiency, and n0r since these parameters are not
-          ! included in lookup table calculations
-
-          ! for T < 273.15, assume all collected rain mass freezes
-          ! note this is a sink for rain mass and number and a source
-          ! for ice mass
-
-          ! note 'f1pr' values are normalized, so we need to multiply by N
-
-          if (qitot_incld(i,k).ge.qsmall .and. qr_incld(i,k).ge.qsmall .and. t(i,k).le.zerodegc) then
-             ! note: f1pr08 and logn0r are already calculated as log_10
-             qrcol = 10._rtype**(f1pr08+logn0r(i,k))*rho(i,k)*rhofaci(i,k)*eri*nitot_incld(i,k)
-             nrcol = 10._rtype**(f1pr07+logn0r(i,k))*rho(i,k)*rhofaci(i,k)*eri*nitot_incld(i,k)
-          endif
-
-          ! for T > 273.15, assume collected rain number is shed as
-          ! 1 mm drops
-          ! note that melting of ice number is scaled to the loss
-          ! rate of ice mass due to melting
-          ! collection of rain above freezing does not impact total rain mass
-
-          if (qitot_incld(i,k).ge.qsmall .and. qr_incld(i,k).ge.qsmall .and. t(i,k).gt.zerodegc) then
-             ! rain number sink due to collection
-             nrcol  = 10._rtype**(f1pr07 + logn0r(i,k))*rho(i,k)*rhofaci(i,k)*eri*nitot_incld(i,k)
-             ! rain number source due to shedding = collected rain mass/mass of 1 mm drop
-             dum    = 10._rtype**(f1pr08 + logn0r(i,k))*rho(i,k)*rhofaci(i,k)*eri*nitot_incld(i,k)
-             ! for now neglect shedding of ice collecting rain above freezing, since snow is
-             ! not expected to shed in these conditions (though more hevaily rimed ice would be
-             ! expected to lead to shedding)
-             !             nrshdr = dum*1.923e+6   ! 1./5.2e-7, 5.2e-7 is the mass of a 1 mm raindrop
-          endif
-
-
+          call ice_drop_collection(rho(i,k),t(i,k),rhofaci(i,k),&
```

Comment:
perhaps something like ice_rain_collection.  Otherwise we have ice_droplet_collection and ice_drop collection and these could be easily confused.

Reply:
Agreed.